api/envoy/config/filter/http/ext_authz/v2/ext_authz.proto

```diff
@@ -90,6 +90,13 @@ message ExtAuthz {
   //
   // If this field is not specified, the filter will be enabled for all requests.
   api.v2.core.RuntimeFractionalPercent filter_enabled = 9;
+
+  // Specifies if the peer certificate is sent to the external service.
+  //
+  // When this field is true, Envoy will include the peer X.509 certificate in the
+  // :ref:`label<envoy_api_field_service.auth.v2.AttributeContext.Peer.labels>` named
+  // `peer-certificate`, in URL- and PEM-encoded format.
+  bool include_peer_certificate = 10;
```

Comment:
I think this is the right thing where the number of items is bounded and built into the Envoy binary. Why not put it in a dedicated peer certificate field as well? In general, I think static typing is much nicer if we can do it, it seems more comprehensible and easier to program with proto idiomatically vs. a label bag.

Reply:
@htuch Something like this?
```diff
diff --git a/api/envoy/config/filter/http/ext_authz/v2/ext_authz.proto b/api/envoy/config/filter/http/ext_authz/v2/ext_authz.proto
index 445c9d7f2..b9359d853 100644
--- a/api/envoy/config/filter/http/ext_authz/v2/ext_authz.proto
+++ b/api/envoy/config/filter/http/ext_authz/v2/ext_authz.proto
@@ -91,12 +91,8 @@ message ExtAuthz {
   // If this field is not specified, the filter will be enabled for all requests.
   api.v2.core.RuntimeFractionalPercent filter_enabled = 9;

-  // Specifies if the peer certificate is sent to the external service.
-  //
-  // When this field is true, Envoy will include the peer X.509 certificate in the
-  // :ref:`label<envoy_api_field_service.auth.v2.AttributeContext.Peer.labels>` named
-  // `peer-certificate`, in URL- and PEM-encoded format.
-  bool include_peer_certificate = 10;
+  // Specifies if and how the peer certificate is sent to the external service.
+  PeerCertificateSettings peer_certificate = 10;
 }

 // Configuration for buffering the request data.
@@ -113,6 +109,14 @@ message BufferSettings {
   bool allow_partial_message = 2;
 }

+// Configuration regarding sending of the peer certificate to the external service.
+message PeerCertificateSettings {
+  // When this field is true, Envoy will include the peer X.509 certificate in the
+  // :ref:`label<envoy_api_field_service.auth.v2.AttributeContext.Peer.labels>` named
+  // `peer-certificate`, in URL- and PEM-encoded format.
+  bool include_with_request = 1;
+}
+
 // HttpService is used for raw HTTP communication between the filter and the authorization service.
 // When configured, the filter will parse the client request and use these attributes to call the
 // authorization server. Depending on the response, the filter may reject or accept the client
```

Reply:
More along the lines of replacing the use of labels in the message that gets sent to `ext_authz` (rather than just the config). I.e. we can have a dedicated field for peer certificate in https://github.com/envoyproxy/envoy/blob/a3bc4734e07528abf898ae176e0ba30196d165cd/api/envoy/service/auth/v2/attribute_context.proto#L42